custom_components/tuya_local/devices/hornbill_y4_smart_lock.yaml

```diff
@@ -0,0 +1,312 @@
+      - id: 73
+        type: string
+        name: remote_pd_setkey_check
+        optional: true
+        persist: false
+        sensitive: true
+  - entity: event
+    name: Alert
+    dps:
+    # TODO: Why not working? PR#2824 also didn't include it.
+    #  - id: 20
+    #    type: string
+    #    name: lock_record
+    #    optional: ?
+      # alarm_lock
+      - id: 21
+        type: string
+        name: event
+        optional: true
+        mapping:
+          - dps_val: wrong_finger
+            value: wrong_finger
+          - dps_val: wrong_password
+            value: wrong_password
+          - dps_val: wrong_card
+            value: wrong_card
+          - dps_val: wrong_face
+            value: wrong_face
+          - dps_val: tongue_bad
+            value: tongue_bad
+          - dps_val: too_hot
+            value: too_hot
+          - dps_val: unclosed_time
+            value: unclosed_time
+          - dps_val: tongue_not_out
+            value: tongue_not_out
+          - dps_val: pry
+            value: pry
+          - dps_val: key_in
+            value: key_in
+          - dps_val: low_battery
+            value: low_battery
+          - dps_val: power_off
+            value: power_off
+          - dps_val: shock
+            value: shock
+          - dps_val: defense
+            value: defense
+          - dps_val: wrong_unlock
+            value: wrong_unlock
+      # doorbell_song
+      - id: 26
+        type: string
+        name: event
+        optional: true
+        mapping:
+          - dps_val: ding_0
+            value: ding_0
+          - dps_val: ding_1
+            value: ding_1
+  - entity: select
+    name: Lock volume
+    category: config
+    dps:
+      - id: 31
+        type: string
+        name: option
+        mapping:
+          - dps_val: mute
+            value: Mute
+          - dps_val: low
+            value: Low
+          - dps_val: normal
+            value: Normal
+          - dps_val: high
+            value: High
+  - entity: switch
+    name: reverse_lock
+    category: config
+    icon: "mdi:screen-lock-rotation"
+    dps:
+      - id: 32
+        type: boolean
+        name: switch
+  - entity: switch
+    name: automatic_lock
+    category: config
+    icon: "mdi:lock-reset"
+    dps:
+      - id: 33
+        type: boolean
+        name: switch
+  - entity: number
+    name: auto_lock_time
+    class: duration
+    icon: "mdi:lock-clock"
+    category: config
+    dps:
+      - id: 36
+        type: integer
+        name: value
+        unit: s
+        range:
+          min: 1
+          max: 1800
+  - entity: select
+    name: closed_opened
+    category: config
+    dps:
+      - id: 40
+        type: string
+        name: option
+        mapping:
+          - dps_val: unknown
+            value: unknown
+          - dps_val: open
+            value: open
+          - dps_val: closed
+            value: closed
+  - entity: select
+    name: autolock_delay_level
```

Comment:
Entity name should be human readable: "Autolock delay..." The name also needs to make clear the difference between this setting and "Auto lock time" above.